manta-pay/src/crypto/constraint/arkworks/mod.rs

```diff
@@ -470,6 +468,19 @@ where
     }
 }
 
+impl<F> BitAnd<Self, R1CS<F>> for Boolean<F>
```

Comment:
I don't understand this. What is the difference between this implementation, which defines a function bitand for Boolean<F>, and the already existing function and? I guess you want to use the fact that Boolean<F> implements the trait BitAnd for some abstractions later on, otherwise I don't see the point of this implementation.

Reply:
This trait is used to define generic circuits which support an `&` operation (called `BitAnd` in Rust). This implementation allows us to write circuits in a generic way without relying on the arkworks backend circuit definition system.